ipfshttpclient/exceptions.py

```diff
@@ -97,27 +102,44 @@ class DecodingError(EncoderError):
 	"""Raised when decoding a byte string to a Python object has failed due to
 	some problem with the input data."""
 	__slots__ = ("original",)
-	#original: Exception
+
+	original: Exception
 	
 	def __init__(self, encoder_name: str, original: Exception) -> None:
-		self.original = original  # type: Exception
+		self.original = original
 		
 		super().__init__("Object decoding error: {}".format(original), encoder_name)
 
 
+##################
+# filescanner.py #
+##################
+
+class MatcherSpecInvalidError(TypeError):
```

Comment:
Make this `class MatcherSpecInvalidError(Error, TypeError)` so that will also be caught when run catch-all blocks for any of the library's exceptions (`try: … except ipfshttpclient.exception.Error: …`)